File: Model/Vehicle.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace asp_dotnet_core_angular.Model
{
    public class Vehicle
    {
        public int Id { get; set; }
        public int ModelId { get; set; }
        public Model Model { get; set; }
        public bool IsRegistered { get; set; }
        public int ContactId { get; set; }
        public Contact Contact { get; set; }
        public DateTime LastUpdated { get; set; }
        public ICollection<VehicleFeature> Features { get; set; }
        public ICollection<Photo> Photos {get;set;}

        public Vehicle()
        {
            Features = new Collection<VehicleFeature>();
            Photos = new Collection<Photo>();
        }
    }
}

File: application/ClientApp/src/app/models/vehicle.ts
export interface Make {
    id: string;
    name: string;
    models:[{id:string,name:string}]
  }
  export interface Feature {
    id: string;
    name: string;
  }
  export interface KeyValuePair{
    id: number;
    name: string;
  }
  
  export interface Contact{
    email: string;
    name: string;
    phone: string;
  }
  
  export interface Vehicle{
    id: number,
    model:KeyValuePair;
    make:KeyValuePair;
    isRegistered:boolean;
    contact: Contact;
    lastUpdated: string;
    features:KeyValuePair[];
    }
  

  export interface QueryResult{
    totalItems:number;
    items: Vehicle[];
  }
  
    
  export interface SaveVehicle{
    id: number;
    modelId:number;
    makeId:number;
    isRegistered:boolean;
    contact: Contact;
    features:number[];
    }

    export interface Photo{
      id:number,
      fileName:string;
    }